packages/fastify-vite/config.js

```diff
@@ -232,31 +234,41 @@ function resolveClientModule(root) {
   return null
 }
 
-function resolveRoot(path) {
+function resolveRoot(distDir, path) {
   let root = path
   if (root.startsWith('file:')) {
     root = fileURLToPath(root)
   }
+  let dir = root
   if (stat(root).isFile()) {
-    return dirname(root)
+    dir = dirname(root)
   }
-  return root
+  // Check if running under dist folder
+  const split = dir.split(sep)
+  if (split.at(-1) === distDir) {
+    return split.slice(0, -1).join(sep)
+  }
+  return dir
 }
 
 async function resolveViteConfig(root, dev, { spa, distDir } = {}) {
   const command = 'build'
   const mode = dev ? 'development' : 'production'
   if (!dev) {
+    let viteDistDir = distDir
+    if (!isAbsolute(viteDistDir)) {
```

Comment:
You can use the `resolveIfRelative` function that is imported on line 7 to do this instead of repeating these 4 lines over and over.

Reply:
We should consider a library like [`pathe`](https://www.npmjs.com/package/pathe) for this. 

Let's leave this for a second round of refactoring focused on path handling alone.